pkg/platform/packet/packet.go

```diff
@@ -428,6 +428,7 @@ func (c *config) checkValidConfig() hcl.Diagnostics {
 	diagnostics = append(diagnostics, c.checkWorkerPoolNamesUnique()...)
 	diagnostics = append(diagnostics, c.checkReservationIDs()...)
 	diagnostics = append(diagnostics, c.validateOSVersion()...)
+	diagnostics = append(diagnostics, c.checkWorkerPoolLabelsAndTaints()...)
```

Comment:
Suggestion to improve commit messages:
- We do not _fix_ anything here, but we add new validation rule, so commit message should reflect that.
- Your commit message does not explain why user cannot have whitespace characters in labels or taints.
- Your commit message only mentiones spaces, while we care about all whitespace characters.
- Commit message should briefly explain what it adds.
- Commit message should include issue reference to make it easier to find context for it, otherwise one needs
  to look for associated merge commit later.

```diff
-Packet: fix labels and taints with spaced value
+packet: add labels and taints validation against whitespace

-Now user cannot have spaced key or value in labels, taints map for
-worker pools.
-Add tests.
+Currently if user adds whitespace characters to labels or taints, it breaks the cluster
+provisioning, as labels and taints are not allowed to contain whitespace.

+To fail early on such cases, this commit adds validation rules, so error is returned to
+ the user before cluster provisioning starts.

+Refs #X
```